hedera-mirror-rest/utils.js

```diff
@@ -733,23 +733,24 @@ const randomString = async (length) => {
 const hexPrefix = '0x';
 const addHexPrefix = (hexData) => {
   if (_.isEmpty(hexData)) {
-    return null;
+    return hexPrefix;
   }
 
   const hexString = typeof hexData === 'string' ? hexData : Buffer.from(hexData).toString();
-  return hexString.indexOf(hexPrefix) === 0 ? hexString : `${hexPrefix}${hexString}`;
+  return hexString.substring(0, 2) === hexPrefix ? hexString : `${hexPrefix}${hexString}`;
 };
 
 /**
  * Converts the byte array returned by SQL queries into hex string
+ * Logic conforms with ETH hex value encoding, therefore nill and empty return '0x'
  * @param {Array} byteArray Array of bytes to be converted to hex string
  * @param {boolean} addPrefix Whether to add the '0x' prefix to the hex string
  * @param {Number} padLength The length to left pad the result hex string
  * @return {String} Converted hex string
  */
 const toHexString = (byteArray, addPrefix = false, padLength = undefined) => {
-  if (_.isNil(byteArray)) {
-    return null;
+  if (_.isNil(byteArray) || _.isEmpty(byteArray)) {
```

Comment:
`isEmpty` also checks if nil.

```suggestion
  if (_.isEmpty(byteArray)) {
```

Reply:
Yeah recalled after pushing, coming in next update along with functionparams catch

Reply:
Updated